justfile

```
@@ -80,27 +80,37 @@ win-arm64-ring16 := if target == "aarch64-pc-windows-msvc" { " --config='patch.c
-rustc-icf := "" #if for-release != "" { " -C link-arg=-Wl,--icf=all" } else { "" }
+# Only enable it on windows where build succeeds, as on other targets
+# it requires the gold linker
+rustc-icf := if for-release != "" { if target-os == "windows" { " -C link-arg=-Wl,--icf=safe" } else { "" } } else { "" }
+
+# lld is the faster linker for lto builds and only slower than mold for non-lto
+# builds.
+#
+# It also supports --icf=safe.
+linker := " -C link-arg=-fuse-ld=lld"
 
 cargo-build-args := (if for-release != "" { " --release" } else { "" }) + (if target != target-host { " --target " + target } else if cargo-buildstd != "" { " --target " + target } else { "" }) + (cargo-buildstd) + (if extra-build-args != "" { " " + extra-build-args } else { "" }) + (cargo-no-default-features) + (cargo-split-debuginfo) + (if cargo-features != "" { " --features " + cargo-features } else { "" }) + (win-arm64-ring16)
 export RUSTFLAGS := (rustc-gcclibs) + (rustc-miropt) + (rustc-icf)
 
 
 # libblocksruntime-dev provides compiler-rt
-ci-apt-deps := if target == "x86_64-unknown-linux-gnu" { "liblzma-dev libzip-dev libzstd-dev"
-    } else if target == "x86_64-unknown-linux-musl" { "musl-tools"
-    } else if target == "aarch64-unknown-linux-gnu" { "g++-aarch64-linux-gnu libc6-dev-arm64-cross binutils binutils-aarch64-linux-gnu libblocksruntime-dev"
-    } else { "" }
+ci-apt-deps := if target == "x86_64-unknown-linux-gnu" { "liblzma-dev libzip-dev libzstd-dev lld"
+    } else if target == "x86_64-unknown-linux-musl" { "musl-tools lld"
+    } else if target == "aarch64-unknown-linux-gnu" { "g++-aarch64-linux-gnu libc6-dev-arm64-cross binutils binutils-aarch64-linux-gnu libblocksruntime-dev lld"
+    } else { "lld" }
 
 [linux]
 ci-install-deps:
-    {{ if ci-apt-deps == "" { "exit" } else { "" } }}
     sudo apt update && sudo apt install -y --no-install-recommends {{ci-apt-deps}}
 
 [macos]
+ci-install-deps:
+    brew update && brew install llvm
```

can write it as

```
    brew update
    brew install llvm
```

`just` implies failing early